leaflet-grids.js

```diff
@@ -0,0 +1,916 @@
+/*
+ *
+ * Inspired by Leaflet.Grid: https://github.com/jieter/Leaflet.Grid
+ */
+
+L.Grids = L.LayerGroup.extend({
+    options: {
+        redraw: 'move',
+        lineStyle: {
+            stroke: true,
+            color: '#111',
+            opacity: 0.6,
+            weight: 1,
+            clickable: false
+        },
+        zoneStyle: {
+                stroke: true,
+                color: '#333',
+                opacity: 0.6,
+                weight: 4,
+                clickable: false
+        },
+    },
+
+    initialize: function (options) {
+        L.LayerGroup.prototype.initialize.call(this);
+        L.Util.setOptions(this, options);
+    },
+
+    onAdd: function (map) {
+        this._map = map;
+        var grid = this.redraw();
+        this._map.on('viewreset ' + this.options.redraw, function () {
+            grid.redraw();
+        });
+        this.eachLayer(map.addLayer, this);
+    },
+
+    redraw: function () {
+        this._lngCoords = [],
+        this._latCoords = [],
+        this._gridLabels =  [],
+        this._mapZoom = this._map.getZoom();
+        this._bounds =  this._map.getBounds(); //.pad(0.5);
+        this._gridSize = this._gridSpacing();
+        this.eachLayer(this.removeLayer, this);
+        var gridLines = this._gridLines();
+        for (i in gridLines){
+            try {
+            this.addLayer(gridLines[i]);
+            }
+            catch (err)
+            {
+                console.log(err);
+                console.log("*******");
+                console.log(gridLines[i]);
+            }
+        }
+
+        for (i in this._gridLabels) {
+            this.addLayer(this._gridLabels[i]);
+        }
+        return this;
+    },
+    
+    _gridSpacing: function () {
+        var zoom = this._map.getZoom();
+        if (zoom > 18) {zoom = 18}
+        return this.options.coordinateGridSpacing[zoom];
+    },
+        
+    _gridLines: function () {
+        var zoom = this._mapZoom; 
+        if(zoom <3){
```

Comment:
Unless there is a good reason for this, i think at least the lat/lon grids should work down to zoom 0.